src/stdlib_linalg_schur.fypp

```diff
@@ -0,0 +1,358 @@
+            call get_schur_${ri}$_workspace(a,lwork,err0)
+            
+            if (err0%error()) then 
+                call linalg_error_handling(err0, err)
+                return
+            else
+                allocate(work(lwork))
+            end if
+            
+        end if        
+        
+        !> SORTING: no sorting options are currently supported
+        sort  = gees_sort_eigs(.false.)
+        sdim  = 0_ilp     
+        
+        if (sort/=GEES_NOT) then 
+            
+           allocate(bwork(n),source=.false.)
+        
+        else
+            
+           bwork => bwork_dummy 
+            
+        end if            
+        
+        !> User or self-allocated eigenvalue storage
+        if (present(eigvals)) then             
+            lde = size(eigvals, 1, kind=ilp)            
+            #:if rt.startswith('c')
+            eigs => eigvals
+            local_eigs = .false.
+            #:else
+            local_eigs = .true.
+            #:endif            
+        else            
+            local_eigs = .true.
+            lde = n            
+        end if
+
+        if (local_eigs) then 
+            ! Use A storage if possible
+            if (overwrite_a_) then                 
+                eigs => a(:,1)
+                #:if rt.startswith('r') 
+                eigi => a(:,2)
+                #:endif
+            else
+                allocate(eigs(n)#{if rt.startswith('r')}#,eigi(n)#{endif}#)
+            end if  
+        endif
+        
+        #:if rt.startswith('c')
+        allocate(rwork(n))
+        #:endif
+
+        if (lde<n) then
+            
+            err0 = linalg_state_type(this, LINALG_VALUE_ERROR, &
+                                           'Insufficient eigenvalue array size=',lde, &
+                                           'should be >=',n)
+        
+        else
+            
+            ! Compute Schur decomposition
+            call gees(jobvs,sort,eig_select,nt,t,mt,sdim,eigs,#{if rt.startswith('r')}#eigi,#{endif}# &
+                      vs,ldvs,work,lwork,#{if rt.startswith('c')}#rwork,#{endif}#bwork,info)
+            call handle_gees_info(info,m,n,m,err0)            
+            
+        
+        end if  
+
+        eigenvalue_output: if (local_eigs) then 
+           #:if rt.startswith('r')
+           ! Build complex eigenvalues
+           if (present(eigvals)) eigvals = cmplx(eigs,eigi,kind=${rk}$)
+           #:endif
+           if (.not.overwrite_a_) deallocate(eigs#{if rt.startswith('r')}#,eigi#{endif}#)
+        endif eigenvalue_output
+        if (.not.present(storage)) deallocate(work)
+        if (sort/=GEES_NOT) deallocate(bwork)
+        call linalg_error_handling(err0,err)
+        
+        contains
+        
+            ! Dummy select routine: currently, no sorting options are offered
+            pure logical(lk) function eig_select(alpha#{if rt.startswith('r')}#r,alphai#{endif}#)             
```

Comment:
I have a possibly silly question, what is the purpose of this dummy function? Given that it just returns `.false.` why not use the value directly in `gees` ? (same for `do_not_select`)

Reply:
`gees` requires an external `procedure(select)` argument: 
```
           ! Function Arguments 
           procedure(stdlib_select_d) :: select
```

Complex and real `select` interfaces are different in LAPACK, hence the templated implementation. If in the future we want to support eigenvalue cherry-picking (i.e. scipy does that), we will do it within this function.

Reply:
Ok, I see, thanks for the clarification!